pallets/otc-settlements/src/lib.rs

```diff
@@ -0,0 +1,544 @@
+	pub enum Event<T: Config> {
+		/// A trade has been executed
+		Executed { asset_id: AssetIdOf<T>, profit: Balance },
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Otc order not found
+		OrderNotFound,
+		/// OTC order is not partially fillable
+		NotPartiallyFillable,
+		/// Provided route doesn't match the existing route
+		InvalidRoute,
+		/// Initial and final balance are different
+		BalanceInconsistency,
+		/// Trade amount higher than necessary
+		TradeAmountTooHigh,
+		/// Trade amount lower than necessary
+		TradeAmountTooLow,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Close an existing OTC arbitrage opportunity.
+		///
+		/// Executes a trade between an OTC order and some route.
+		/// If the OTC order is partially fillable, the extrinsic fails if the existing arbitrage
+		/// opportunity is not closed after the trade.
+		/// If the OTC order is not partially fillable, fails if there is no profit after the trade.
+		///
+		/// `Origin` calling this extrinsic is not paying or receiving anything.
+		///
+		/// The profit made by closing the arbitrage is transferred to `FeeReceiver`.
+		///
+		/// Parameters:
+		/// - `origin`: Signed or unsigned origin. Unsigned origin doesn't pay the TX fee,
+		/// 			but can be submitted only by a collator.
+		/// - `otc_id`: ID of the OTC order with existing arbitrage opportunity.
+		/// - `amount`: Amount necessary to clone the arb.
+		/// - `route`: The route we trade against. Required for the fee calculation.
+		///
+		/// Emits `Executed` event when successful.
+		///
+		#[pallet::call_index(0)]
+		#[pallet::weight(<T as Config>::WeightInfo::settle_otc_order()
+			.saturating_add(<T as Config>::RouterWeightInfo::sell_weight(route))
+		.saturating_add(<T as Config>::RouterWeightInfo::get_route_weight())
+		.saturating_add(<T as Config>::RouterWeightInfo::calculate_spot_price_with_fee_weight(route))
+		.saturating_add(<T as pallet_otc::Config>::WeightInfo::fill_order().max(<T as pallet_otc::Config>::WeightInfo::partial_fill_order()))
+		)]
+		pub fn settle_otc_order(
+			_origin: OriginFor<T>,
+			otc_id: OrderId,
+			amount: Balance,
+			route: Vec<Trade<AssetIdOf<T>>>,
+		) -> DispatchResult {
+			Self::settle_otc(otc_id, amount, route)
+		}
+	}
+}
+
+impl<T: Config> Pallet<T> {
+	pub fn account_id() -> T::AccountId {
+		PALLET_ID.into_account_truncating()
+	}
+
+	/// Ensure that the profit is more than some minimum amount.
+	fn ensure_min_profit(otc_amount_in: Balance, _profit: Balance) -> DispatchResult {
+		let _min_expected_profit = T::MinProfitPercentage::get().mul_floor(otc_amount_in);
+
+		// In the benchmark we calculate the overhead of extrinsic and we doesn't make any trade.
+		// We disable this check because otherwise it would fail.
+		#[cfg(not(feature = "runtime-benchmarks"))]
+		// tell the binary search algorithm to find higher values
+		ensure!(_profit >= _min_expected_profit, Error::<T>::TradeAmountTooLow);
+
+		Ok(())
```

Comment:
nitpick: i believe what you can do to make it more readable
```
                // In the benchmark we calculate the overhead of extrinsic and we doesn't make any trade.
		// We disable this check because otherwise it would fail.
                if cfg!(feature = "runtime-benchmarks") {
                    return Ok(());
                }
                
		let min_expected_profit = T::MinProfitPercentage::get().mul_floor(otc_amount_in);
		// tell the binary search algorithm to find higher values
		ensure!(profit >= min_expected_profit, Error::<T>::TradeAmountTooLow);
		Ok(())
```